src/Microsoft.AspNet.Diagnostics/StatusCodePagesMiddleware.cs

```diff
@@ -25,8 +25,17 @@ public StatusCodePagesMiddleware(RequestDelegate next, [NotNull]StatusCodePagesO
 
         public async Task Invoke(HttpContext context)
         {
+            context.SetFeature<IStatusCodeFeature>(new StatusCodeFeature());
+
             await _next(context);
 
+            var statusCodeFeature = context.GetFeature<IStatusCodeFeature>();
+            if (statusCodeFeature != null && !statusCodeFeature.Enabled)
```

Comment:
Since this middleware is setting this feature, shouldn't it be considered off if it is not around?